src/internal/m365/collection/drive/limiter.go

```diff
@@ -81,25 +74,58 @@ func (l pagerLimiter) atContainerItemsLimit(numItems int) bool {
 	return l.enabled() && numItems >= l.limits.MaxItemsPerContainer
 }
 
-// atContainerPageLimit returns true if the limiter is enabled and the number of
+// atPageLimit returns true if the limiter is enabled and the number of
 // pages processed so far is beyond the limit for this backup.
 func (l pagerLimiter) atPageLimit(stats *driveEnumerationStats) bool {
 	return l.enabled() && stats.numPages >= l.limits.MaxPages
 }
 
 // atLimit returns true if the limiter is enabled and meets any of the
 // conditions for max items, containers, etc for this backup.
-func (l pagerLimiter) atLimit(
-	stats *driveEnumerationStats,
-	containerCount int,
-) bool {
-	nc := stats.numContainers
-	if nc == 0 && containerCount > 0 {
-		nc = containerCount
-	}
-
+func (l pagerLimiter) atLimit(stats *driveEnumerationStats) bool {
 	return l.enabled() &&
 		(l.atItemLimit(stats) ||
-			nc >= l.limits.MaxContainers ||
+			stats.numContainers >= l.limits.MaxContainers ||
 			stats.numPages >= l.limits.MaxPages)
 }
+
+// ---------------------------------------------------------------------------
+// Used by the tree version limit handling
+// ---------------------------------------------------------------------------
+
+// hitLimit returns true if the limiter is enabled and meets any of the
+// conditions for max items, containers, etc for this backup.
+func (l pagerLimiter) hitLimit(
+	pageCount, containerCount, itemCount int,
+	totalBytesCount int64,
```

Comment:
I'd much prefer keeping these values in a struct so that it's both easy to update the set of things that's being checked without having to go update a bunch of function signatures and so that it's easier for the dev calling the function because they don't need to memorize/lookup the order of parameters each time, especially since so many of them are the same type. This is somewhat similar to the reasons we've been trying to move to returning structs instead of repeated values of the same type

Reply:
This func gets deleted in the next PR, since I realized it wasn't getting used.  This version of it came from an assumption of need that didn't pan out.  As for keeping counts owned by a single struct, I think you'll be happy with the implementation in the next PR.